administrator/components/com_contact/src/View/Contacts/HtmlView.php

```diff
@@ -143,7 +143,10 @@ public function display($tpl = null)
      */
     protected function addToolbar()
     {
-        $canDo = ContentHelper::getActions('com_contact', 'category', $this->state->get('filter.category_id'));
+        // Get allowed actions form the component or category (but only when filtering is by single category)
```

Comment:
```suggestion
        // Get allowed actions from the component or category (but only when filtering is by single category)
```
Repeat with other instances.

Reply:
done